src/zimcheck.cpp

```diff
@@ -396,28 +396,35 @@ void test_articles(const zim::File& f, ErrorLogger& reporter, ProgressBar progre
             baseUrl.resize( pos==baseUrl.npos ? 0 : pos );
 
             auto links = getLinks(it->getData());
-            for(auto olink: links)
+            std::vector<std::string> filtered;
+            for (const auto &l : links)
             {
-                if (olink.front() == '#' || olink.front() == '?')
-                    continue;
-                if (isInternalUrl(olink)) {
-                    auto link = normalize_link(olink, baseUrl);
-                    char nm = link[0];
-                    std::string shortUrl(link.substr(2));
-                    auto a = f.getArticle(nm, shortUrl);
-                    if (!a.good())
+                if (l.front() == '#') continue;
+                if (isInternalUrl(l) == false) continue;
+
+                filtered.push_back(normalize_link(l, baseUrl));
+            }
+
+            // remove duplicates
+            sort(filtered.begin(), filtered.end());
+            filtered.erase(unique(filtered.begin(), filtered.end()), filtered.end());
+            for(const auto &olink: filtered)
+            {
+                char nm = olink[0];
```

Comment:
The "o" of `olink` was probably here for "original link".
This is no more the original link as it is the normalized link.
`link` is a better name.